convert_test.go

```diff
@@ -90,8 +91,29 @@ func TestFromTime(base *testing.T) {
 			Error: "timestamp contains more than just date information; 2020-05-11T07:10:55.209309302Z",
 		},
 		{
-			Time:  "2022-01-31T00:00:00.000-05:00",
-			Error: "timestamp contains more than just date information; 2022-01-31T00:00:00-05:00",
+			Time:     "2022-01-31T00:00:00.000-05:00",
+			Timezone: timezoneMetadata{Name: valueToPtr(""), Offset: valueToPtr(-18000)},
+			Error:    "timestamp contains more than just date information; 2022-01-31T00:00:00-05:00",
+		},
+		{
+			Time:     "2022-01-31T05:00:00.000Z",
+			Timezone: timezoneMetadata{InTimezone: valueToPtr("America/New_York"), Name: valueToPtr("EST"), Offset: valueToPtr(-18000)},
```

Comment:
This is an explicitly **NEW** tested case, i.e. a **NAMED** timezone. Parsing `"2022-01-31T00:00:00.000-05:00"` on a machine which isn't in that local timezone (e.g. not `TZ=America/New_York`) would be equivalent to a `time.FixedZone("", -18000)`, i.e. not a named timezone.